src/stub/src/powerpc64le-linux.elf-main.c

```diff
@@ -136,13 +136,16 @@ ERR_LAB
         //   assert(h.sz_cpr > 0 && h.sz_cpr <= blocksize);
 
         if (h.sz_cpr < h.sz_unc) { // Decompress block
-            long out_len = h.sz_unc;  // EOF for lzma
+            uint64_t out_len = h.sz_unc;  // EOF for lzma
             int const j = (*f_decompress)((const unsigned char *)xi->buf, h.sz_cpr,
                 (unsigned char *)xo->buf, &out_len, h.b_method);
-            if (j != 0 || out_len != (nrv_uint)h.sz_unc)
+            if (j != 0 || out_len != (uint64_t)h.sz_unc)
                 err_exit(7);
-            if (h.b_ftid!=0 && f_unf) {  // have filter
-                (*f_unf)((unsigned char *)xo->buf, out_len, h.b_cto8, h.b_ftid);
+            if (h.b_ftid!=0 && f_unf
+               &&  ((512 < out_len)  // this block is longer than Ehdr+Phdrs
```

Comment:
Please insert a multi-sentence explanation of the '&&' and "||' on these two lines, about whether to apply f_unf or not.  I'm not convinced by the short remarks.